src/visor/historydb/historydb.go

```diff
@@ -56,9 +60,79 @@ func New(db *bolt.DB) (*HistoryDB, error) {
 		return nil, err
 	}
 
+	hd.addrTxns, err = newAddressTxnsBkt(db)
+	if err != nil {
+		return nil, err
+	}
+
 	return &hd, nil
 }
 
+// ResetIfNeed checks if need to reset the parsed block history,
+// If we have a new added bucket, we need to reset to parse
+// blockchain again to get the new bucket filled.
+func (hd *HistoryDB) ResetIfNeed() error {
+	if hd.historyMeta.ParsedHeight() == 0 {
+		return nil
+	}
+
+	// if any of the following buckets are empty, need to reset
+	var reset bool
+	for {
```

Comment:
how is the for loop different from `if hd.addrTxns.IsEmpty() || hd.addrUx.IsEmpty() || hd.txns.IsEmpty() || hd.outputs.IsEmpty() { return hd.reset() }`